Cgen error on multi return using generic reference

### Describe the bug

```
==================
/dev/shm/v_1000/ttt.01HZ2AVGWZ3T09RBVJKJS01YQP.tmp.c:13147: error: cannot convert 'struct main__Asset' to 'struct main__Asset *'
...
==================
```

### Reproduction Steps

```v
module main

pub enum AssetGetStatus {
    ok
    error
}

struct Sound {}

struct Asset {}

pub fn get[T]() (T, AssetGetStatus) {
    $if T is Sound {
        return Sound{}, AssetGetStatus.ok
    } $else $if T is &Asset {
        return &Asset{}, AssetGetStatus.ok
    } $else {
        $compile_error('get[T]: only retreival of Sound and &Asset is currently supported')
    }
    // Should not could be reached:
    return T{}, AssetGetStatus.error
}

fn main() {
    asset, status := get[&Asset]()
    dump(asset)
    dump(status)
}
```

### Expected Behavior

Clean compile and dump output

### Current Behavior

```
==================
/dev/shm/v_1000/ttt.01HZ2AVGWZ3T09RBVJKJS01YQP.tmp.c:13147: error: cannot convert 'struct main__Asset' to 'struct main__Asset *'
...
==================
```

### Possible Solution

The `&Asset` could be wrapped in a stack-based type, e.g: `AssetRef` but it is a little cumbersome to work with manual wrapping and unwrapping like that :slightly_smiling_face: 

### Additional Information/Context

_No response_

### V version

V 0.4.6 09eaae5

### Environment details (OS name and version, etc.)

Linux, EndeavourOS (rolling, Arch based)

--- a/vlib/v/gen/c/struct.v
+++ b/vlib/v/gen/c/struct.v
@@ -115,7 +115,8 @@ fn (mut g Gen) struct_init(node ast.StructInit) {
 		}
 	} else {
 		// alias to pointer type
-		if g.table.sym(node.typ).kind == .alias && g.table.unaliased_type(node.typ).is_ptr() {
+		if (g.table.sym(node.typ).kind == .alias && g.table.unaliased_type(node.typ).is_ptr())
+			|| (node.typ.has_flag(.generic) && g.unwrap_generic(node.typ).is_ptr()) {
 			g.write('&')
 		}
 		if is_array || const_msvc_init {

--- a/vlib/v/tests/generic_struct_init_ptr_test.v
+++ b/vlib/v/tests/generic_struct_init_ptr_test.v
@@ -1,0 +1,27 @@
+module main
+
+pub enum AssetGetStatus {
+	ok
+	error
+}
+
+struct Sound {}
+
+struct Asset {}
+
+pub fn get[T]() (T, AssetGetStatus) {
+	$if T is Sound {
+		return Sound{}, AssetGetStatus.ok
+	} $else $if T is &Asset {
+		return &Asset{}, AssetGetStatus.ok
+	} $else {
+		$compile_error('get[T]: only retreival of Sound and &Asset is currently supported')
+	}
+	// Should not could be reached:
+	return T{}, AssetGetStatus.error
+}
+
+fn test_main() {
+	asset, status := get[&Asset]()
+	assert status == AssetGetStatus.ok
+}

